packages/custom-storage/src/StorageRepository.ts

```diff
@@ -0,0 +1,37 @@
+import { ColumConfig } from '@carto/toolkit-sql/dist/types/DDL';
+
+export interface StoredVisualization extends Visualization {
+  id: string;
+}
+
+export interface CompleteVisualization {
+  vis: StoredVisualization;
+  datasets: Dataset[];
+}
+
+export interface Visualization {
+  name: string;
+  desc: string;
+  img: string;
+  isPrivate: boolean;
+  config: string;
+}
+
+export interface Dataset {
+  name: string;
+  columns?: Array<string | ColumConfig>;
+  file: string;
+}
+
+export interface StorageRepository {
+  getVisualizations(): Promise<StoredVisualization[]>;
+  getPublicVisualizations(): Promise<StoredVisualization[]>;
+  getPrivateVisualizations(): Promise<StoredVisualization[]>;
+  getVisualization(id: string): Promise<CompleteVisualization | null>;
+  getPublicVisualization(id: string): Promise<CompleteVisualization | null>;
+  deleteVisualization(id: string): Promise<boolean>;
+  createVisualization(visualization: Visualization, datasets: Dataset[]): Promise<any>;
+  updateVisualization(visualization: StoredVisualization, datasets: Dataset[]): Promise<any>;
```

Comment:
A general question about 'StoredVisualization' and other interfaces referring to different states. 

What's the main advantage you see over using just a plain class `Visualization` with all possible fields, some of them null or empty and some check XXX methods?

Reply:
Less code and less state management, mainly.

Also I think that, at a glance, you can see that they're not exactly the same. If both signatures were visualization: Visualization, you might think that you can specify the id on `create`, when you really shouldn't. However, you absolutely need to have it on `update`

Reply:
Ok, let's see how that works 👌